linalg/hypre.cpp

```diff
@@ -137,10 +137,18 @@ HypreParVector::HypreParVector(const HypreParVector &y) : Vector()
 #endif
    hypre_ParVectorSetDataOwner(x,1);
    hypre_SeqVectorSetDataOwner(hypre_ParVectorLocalVector(x),1);
+   // Deep copy the local data
+   hypre_SeqVectorCopy(hypre_ParVectorLocalVector(y.x),
+                       hypre_ParVectorLocalVector(x));
    _SetDataAndSize_();
    own_ParVector = 1;
 }
 
+HypreParVector::HypreParVector(HypreParVector &&y)
```

Comment:
```suggestion
HypreParVector::HypreParVector(HypreParVector &&y) : HypreParVector()
```
My previous suggestion caused the CI failure on Mac (`WrapHypreParVector` was trying to destroy `x` before it initialized). Sorry about that! This should be better now.

Reply:
Yep - I noticed this as well and just set `own_parVector` to false before the move assignment operator call

Reply:
Yeah, that will fix it too. Go with whichever option you think is clearer.